src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -0,0 +1,40 @@
+/*AnjuliJ*/
+package codeu.controller;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Servlet class responsible for user registration
+ */
+
+public class RegisterServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+        throws IOException, ServletException{
+        request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+        //response.getWriter().println("<h1>RegisterServlet GET request.</h1>");
+    }
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException{
+        String username = request.getParameter("username");
+        String password = request.getParameter("password");
+
+        if(!username.matches("[\\w*\\s*]*")){
+            request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
```

Comment:
spaces and word characters seems weird.  Like Bob\tSmith and Bob Smith are both allowed, as is Bob\nSmith.

Also, this regex will pass if username is empty.

Also: " Bob" and "Bob" and "Bob " are all distinct users according to this; do you want to tighten down the whitespace requirements, or maybe get rid of whitespace altogether?

Sorry. There's a saying about regexes.  You have a problem, so you solve it with a regex. Now you have two problems.  (Actually, I think a regex is fine here -- it just should be something like \w[\w\d]*)

What about constraints on password (eg, checking if empty?)